src/hypothesis/strategies.py

```diff
@@ -529,43 +529,35 @@ def lists(
             def unique_by(x):
                 return x
 
+    if min_size is None:
+        min_size = 0
+
+    if max_size is None:
+        max_size = float('inf')
+
+    if average_size is None:
+        average_size = max(
+            _AVERAGE_LIST_LENGTH,
+            min_size * 2
+        )
+        if max_size < float('inf'):
+            average_size = min(average_size, 0.5 * (min_size + max_size))
+
+    from hypothesis.searchstrategy.collections import ListStrategy, \
+        UniqueListStrategy
     if unique_by is not None:
-        from hypothesis.searchstrategy.collections import UniqueListStrategy
-        min_size = min_size or 0
-        max_size = max_size or float(u'inf')
-        if average_size is None:
-            if max_size < float(u'inf'):
-                if max_size <= 5:
-                    average_size = min_size + 0.75 * (max_size - min_size)
-                else:
-                    average_size = (max_size + min_size) / 2
-            else:
-                average_size = max(
-                    _AVERAGE_LIST_LENGTH,
-                    min_size * 2
-                )
-        result = UniqueListStrategy(
+        return UniqueListStrategy(
             elements=elements,
             average_size=average_size,
             max_size=max_size,
             min_size=min_size,
             key=unique_by
         )
     else:
-        from hypothesis.searchstrategy.collections import ListStrategy
-        if min_size is None:
-            min_size = 0
-        if average_size is None:
-            if max_size is None:
-                average_size = _AVERAGE_LIST_LENGTH
-            else:
-                average_size = (min_size + max_size) * 0.5
-
-        result = ListStrategy(
+        return ListStrategy(
```

Comment:
Why does ListStrategy have `average_length` but UniqueListStrategy use `average_size`? 🤔

Reply:
Probably predates our API guidelines, but since it's private API we should probably change it to use `average_size` too.